src/instantsend/instantsend.cpp

```diff
@@ -1639,8 +922,30 @@ size_t CInstantSendManager::GetInstantSendLockCount() const
 void CInstantSendManager::WorkThreadMain(PeerManager& peerman)
 {
     while (!workInterrupt) {
-        bool fMoreWork = ProcessPendingInstantSendLocks(peerman);
-        ProcessPendingRetryLockTxs();
+        bool fMoreWork = [&]() -> bool {
+            if (!IsInstantSendEnabled()) return false;
+            const bool more_work{ProcessPendingInstantSendLocks(peerman)};
+            if (!m_signer) return more_work;
+            // Construct set of non-locked transactions that are pending to retry
+            std::vector<CTransactionRef> txns{};
+            {
+                LOCK2(cs_nonLocked, cs_pendingRetry);
+                if (pendingRetryTxs.empty()) return more_work;
+                txns.reserve(pendingRetryTxs.size());
+                for (const auto& txid : pendingRetryTxs) {
+                    if (auto it = nonLockedTxs.find(txid); it != nonLockedTxs.end()) {
+                        const auto& [_, tx_info] = *it;
+                        if (tx_info.tx) {
+                            txns.push_back(tx_info.tx);
+                        }
+                    }
+                }
+                txns.shrink_to_fit();
```

Comment:
we shouldn't shrink this. It may re-allocate which is ~ expensive. I'd keep the reserve, that's good to avoid reallocations during growth. But as this is short-lived and released after ProcessPendingRetryLockTxs is called, there's no need to shrink it down to save a few bytes of heap usage for such a short time frame.

Reply:
Resolved in latest push.